services/datamanager/datasync/sync.go

```diff
@@ -93,29 +97,41 @@ func NewManager(logger golog.Logger, partID string, client v1.DataSyncServiceCli
 ) (Manager, error) {
 	cancelCtx, cancelFunc := context.WithCancel(context.Background())
 	ret := syncer{
+		partID:             partID,
 		conn:               conn,
 		client:             client,
 		logger:             logger,
 		backgroundWorkers:  sync.WaitGroup{},
 		cancelCtx:          cancelCtx,
 		cancelFunc:         cancelFunc,
-		partID:             partID,
+		lastModifiedMillis: lastModifiedMillis,
 		progressLock:       &sync.Mutex{},
 		inProgress:         make(map[string]bool),
-		lastModifiedMillis: lastModifiedMillis,
+
+		syncErrs:   make(chan error, 10),
+		closed:     &atomic.Bool{},
+		logRoutine: sync.WaitGroup{},
 	}
+	ret.logRoutine.Add(1)
+	goutils.PanicCapturingGo(func() {
```

Comment:
I was initially going to comment that I was wondering whether a buffer size of 10 was sufficient since you could be syncing many, many more files that each have errors and thus block logSyncErrs.

But since you're immediately kicking off `logSyncErrs()`, verifying that it's immediately reading from the `syncErrs` channel and thus keeping the capacity freed up?

Reply:
Yeah exactly - as long as the rate the channel is being drained is > it's being filled, we should be good here. That would only be the case if we're encountering errors faster than we can log (which is a very light op), so if that's ever the case I think we're already in a completely broken state